### PR TITLE
Only search when the search bar is applied via a button click or enter

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
#13

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-data-manager?_sp.spa=pr-search-only-on-apply](https://developer.blackbaud.com/skyux-data-manager?_sp.spa=pr-search-only-on-apply)

File: src/app/public/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts [hunk 1]
```diff
@@ -91,6 +97,38 @@ describe('SkyDataManagerToolbarComponent', () => {
   let modalServiceInstance: MockModalService;
   let viewConfig: SkyDataViewConfig;
 
+  function setSearchInput(text: string) {
+    let inputEvent = document.createEvent('Event');
+    let params = {
+      bubbles: false,
+      cancelable: false
+    };
+    inputEvent.initEvent('input', params.bubbles, params.cancelable);
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts:106]: Would `SkyAppTestUtility.fireDomEvent()` help make this easier?

File: src/app/public/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts [hunk 2]
```diff
@@ -91,6 +97,38 @@ describe('SkyDataManagerToolbarComponent', () => {
...
+    dataManagerToolbarFixture.detectChanges();
+  }
+
+  function triggerSearchInputEnter() {
+    let inputEl = dataManagerToolbarFixture.debugElement.query(By.css('.sky-search-container input'));
+    inputEl.triggerEventHandler('keyup', { which: 13});
+    dataManagerToolbarFixture.detectChanges();
+  }
+
+  function triggerSearchApplyButton() {
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts:126]: Functions need a return type.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-data-manager?_sp.spa=pr-search-only-on-apply](https://developer.blackbaud.com/skyux-data-manager?_sp.spa=pr-search-only-on-apply)

File: src/app/public/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts:106]: Hmmm. Potentially - I honestly copied this in from another spec. That's a good call though - I'll look at it :)

Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts]: Fixed. I did have to cast the keyboard event but I tried to explain why in a code comment in that spot.

File: src/app/public/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/data-manager/data-manager-toolbar/data-manager-toolbar.component.spec.ts]: Fixed all three of the helper functions here.

**Review by Blackbaud-AlexKingman — DISMISSED** [review]

**Review by Blackbaud-AlexKingman — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-data-manager?_sp.spa=pr-search-only-on-apply](https://developer.blackbaud.com/skyux-data-manager?_sp.spa=pr-search-only-on-apply)

**Pull request merged**